webapp/channels/src/components/admin_console/team_channel_settings/group/index.ts

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {connect} from 'react-redux';
+import {ActionCreatorsMapObject, bindActionCreators, Dispatch} from 'redux';
+import {ActionFunc} from 'mattermost-redux/types/actions';
+import {getGroups} from 'mattermost-redux/actions/groups';
+import {t} from 'utils/i18n';
+import {Group} from '@mattermost/types/groups';
+import List from './group_list';
+import {GlobalState} from 'types/store';
+
+type Actions = {
+
+    getData(): Promise<Group[]>;
+}
+type OwnProps = {
+    groups: Group[];
+    totalGroups: number;
+    isModeSync: boolean;
+    onGroupRemoved: (gid: string) => void;
+    setNewGroupRole: (gid: string) => void;
+}
+
+function mapStateToProps(state: GlobalState, ownProps: OwnProps) {
+    return {
+        data: ownProps.groups,
+        total: ownProps.totalGroups,
+        emptyListTextId: ownProps.isModeSync ? t('admin.team_channel_settings.group_list.no-synced-groups') : t('admin.team_channel_settings.group_list.no-groups'),
+        emptyListTextDefaultMessage: ownProps.isModeSync ? 'At least one group must be specified' : 'No groups specified yet',
+        removeGroup: ownProps.onGroupRemoved,
+        setNewGroupRole: ownProps.setNewGroupRole,
+    };
+}
+
+function mapDispatchToProps(dispatch: Dispatch) {
+    return {
+        actions: bindActionCreators<ActionCreatorsMapObject<ActionFunc>, Actions>({
+            getData: () => getGroups(),
```

Comment:
`getGroups` was not used in the older version. Why do we need to use it here or am I missing something?

Reply:
I was getting compilation errors with the older version of the function. I guessed that using  `getGroups` action would give us the same result as `Promise.resolve()`. Do you still want to retain the older version of this function?

Reply:
I dont think `getGroups` is the correct action to be used here. Because if you check `GroupList` action is passed down to `AbstractList` component and `getGroup` type definition is different there. 

In the meantime I am also wondering how this component is working properly with  `Promise.resolve()`. 

Therefore I would request another review from the internal member who knows this area of the code base cc: @M-ZubairAhmed

Reply:
@M-ZubairAhmed any idea here?

Reply:
@M-ZubairAhmed, would really appreciate your insights about the most suitable action for getting groups. This is my first time contributing to the Mattermost codebase

Reply:
Oops sorry i dont know how i missed this notification for so long. Let me answer you tomorrow

Reply:
lets use the same promise.resolve, i think the component should have been properly added the fallback when getData was not defined.